### PR TITLE
Add monitor config-key exists/get/set/rm support

### PR DESCRIPTION
**Pull request by cholcombe973**
I added a few more mon commands.

### CODE REVIEW

File: src/cmd.rs [hunk 1]
```diff
@@ -566,6 +566,75 @@ pub fn cluster_health(cluster_handle: &Rados) -> RadosResult<ClusterHealth> {
...
+        Some(val) => Ok(val.to_string()),
+        None => Err(RadosError::Error(format!(
+            "Unable to parse config-key get output: {:?}",
+            return_data,
+        ))),
+    }
+}
+
+/// Remove a given configuration key from the monitor cluster
+pub fn get_config_remove(cluster_handle: &Rados, key: &str, simulate: bool) -> RadosResult<()> {
```
Comment (ChrisMacNaughton) [inline src/cmd.rs]: `pub fn remove_config` maybe, instead of `get_config_remove`?

File: src/cmd.rs [hunk 2]
```diff
@@ -566,6 +566,75 @@ pub fn cluster_health(cluster_handle: &Rados) -> RadosResult<ClusterHealth> {
...
+    });
+
+    if !simulate {
+        cluster_handle.ceph_mon_command_without_data(&cmd)?;
+    }
+    Ok(())
+}
+
+/// Set a given configuration key in the monitor cluster
+pub fn get_config_set(cluster_handle: &Rados, key: &str, value: &str, simulate: bool) -> RadosResult<()> {
```
Comment (ChrisMacNaughton) [inline src/cmd.rs]: `pub fn set_config` maybe?

**Review by ChrisMacNaughton — CHANGES_REQUESTED** [review]
Some minor requests on function names

File: src/cmd.rs (hunk 1 above)
Comment (cholcombe973) [reply src/cmd.rs]: Good idea!  I'll rename these

File: src/cmd.rs (hunk 2 above)
Comment (cholcombe973) [reply src/cmd.rs]: Yup i'll rename it

File: src/cmd.rs [hunk 3]
```diff
@@ -566,6 +566,75 @@ pub fn cluster_health(cluster_handle: &Rados) -> RadosResult<ClusterHealth> {
     Ok(serde_json::from_str(&return_data)?)
 }
 
+/// Check with the monitor whether a given key exists
+pub fn get_config_exists(cluster_handle: &Rados, key: &str) -> RadosResult<bool> {
```
Comment (ChrisMacNaughton) [inline src/cmd.rs]: One more nit: maybe `pub fn config_key_exists()`

**Comment by cholcombe973** [comment]
@sdandam.  This is for the journal support.  These functions are needed to grab the currently set journal size when creating a journal

**Comment by cholcombe973** [comment]
Ok I tested these and they work.  Unfortunately I ran into a problem.  config-key exists uses the status message to indicate if a key exists instead of the usual mechanism.  This caused me to have to expose the status message which is a breaking change.  I tried to use the existing ceph_mon_command_with_data function but even passing a blank vec! into that causes ceph to error out with a -22 error which means invalid argument.  So the function doesn't take data but I need to expose that status string :-/

**Comment by cholcombe973** [comment]
@sdandam turns out I was wrong and that the function I needed was in the library all along.  These are still good to have for completeness

**Pull request merged**